src/hotspot/share/nmt/nmtNativeCallStackStorage.hpp

```diff
@@ -41,19 +42,17 @@
 // We achieve this by using a closed hashtable for finding previously existing NCS:s and referring to them by an index that's smaller than a pointer.
 class NativeCallStackStorage : public CHeapObjBase {
 public:
-  struct StackIndex {
-    friend NativeCallStackStorage;
-    int32_t _stack_index;
-  public:
-    static constexpr const int32_t invalid = -1;
-    static bool equals(const StackIndex& a, const StackIndex& b) {
-      return a._stack_index == b._stack_index;
-    }
+  using StackIndex = int;
 
-    bool is_invalid() {
-      return _stack_index == invalid;
-    }
-  };
+  constexpr static const StackIndex invalid = std::numeric_limits<StackIndex>::max() - 1;
```

Comment:
I'm trying to figure out whether this needs to be `public` and even though I see other classes using `StackIndex`, I don't see them actually needing to check the value.

I think it's fine to leave it `public`, though I would have started with `private` and made it `public` only if strictly needed.

Though I'd probably prefer if it was called `InvalidIndexValue` or something better than just `invalid`.

Reply:
Yeah, we can just leave it private and force a PR which needs it to make it public. Let's keep the simple name for now, we'll see how readable it is when/if it becomes a public field.